### ISSUE TITLE
Supervisord Segmentation Fault

### ISSUE THREAD

**Issue by akurniawan** [Supervisor/supervisor#890]
Hi all,

This morning we were experiencing failure in supervisord processes. For unknown reason supervisord was killed and threw these errors on kern.log

> Feb 14 01:03:45 metricsdata-be-03 kernel: 24234.380489 supervisord1459: segfault at 92 ip 0000000000490964 sp 00007ffc6da9e670 error 4 in python[400000+2bc000]
Feb 14 01:03:45 metricsdata-be-03 kernel: 24234.453256 python12030: segfault at 0 ip (null) sp 00007ffb7d843488 error 14 in python2.7[400000+2bc000]

This is the configuration we use for supervisord

> ; supervisor config file
[unix_http_server]
file=/opt/supervisor/run/supervisor.sock
[supervisord]
logfile=/opt/supervisor/log/supervisord.log
pidfile=/opt/supervisor/run/supervisord.pid
[rpcinterface:supervisor]
supervisor.rpcinterface_factory = supervisor.rpcinterface:make_main_rpcinterface
[supervisorctl]
serverurl=unix:///opt/supervisor/run/supervisor.sock
[include]
files = /etc/supervisor/conf.d/*.conf

Below are the list of software we manage with supervisord and its configuration

> [program:carbon-cache]
command=/usr/bin/python /opt/graphite/bin/carbon-cache.py --instance=%(process_num)s --nodaemon start
process_name=%(program_name)s_%(process_num)02d
numprocs=12
user=root
stdout_logfile = /var/log/graphite/carbon-cache.log
stdout_logfile_maxbytes = 100MB
stdout_logfile_backups = 5
stderr_logfile = /var/log/graphite/carbon-cache-err.log
stderr_logfile_maxbytes = 100MB
stderr_logfile_backups = 5

> [program:carbon-relay]
command=/usr/bin/python /opt/graphite/bin/carbon-relay.py --instance=%(process_num)s --nodaemon start
process_name=%(program_name)s_%(process_num)02d
numprocs=1
user=root
stdout_logfile = /var/log/graphite/carbon-relay.log
stdout_logfile_maxbytes = 100MB
stdout_logfile_backups = 5
stderr_logfile = /var/log/graphite/carbon-relay-err.log
stderr_logfile_maxbytes = 100MB
stderr_logfile_backups = 5

> [program:graphite-web]
command = gunicorn -w16 graphite_wsgi -k gevent -b 0.0.0.0:8888 --access-logfile /var/log/graphite/graphite.log --error-logfile /var/log/graphite/graphite-error.log --log-level info --timeout 420
autostart = true
autorestart = true
directory = /opt/graphite/webapp/graphite
user = root

Do you guys have any insight what may causing this problem?

**Comment by mnaberez** [comment]
`supervisord` is a pure Python program.  It only has one dependency, meld3, which is also our package and is a pure Python program.  Being pure Python code (no C extensions that access memory directly), `supervisord` itself shouldn't be able to cause a segmentation fault.  I think it's more likely the segfault is caused by a bad build of Python, something else weird about the system this error occurred on, or maybe a bug in Python itself.

Since we haven't had any similar reports and it shouldn't be possible for `supervisord` itself to segfault, I'm going to close this issue.  If you find evidence that `supervisord` itself really is the cause and post it here, we can reopen.

**Comment by jasrys** [comment]
I too am getting a segfault (looks like something related to `asynchat_25.py`?) with supervisor on Ubuntu 16.04/PHP 7.1. I tried on both supervisor 3.2 that is available via apt-get as well as 3.3.1 via pip and experienced segfaults in both versions. Any ideas would be much appreciated. Thanks!

From syslog:

```
Apr 10 08:24:51 konker2 kernel: [246856.009468] php[19731]: segfault at 700000001 ip 000055d52fb520a2 sp 00007fff3bec7158 error 4 in php7.1[55d52f90c000+3d2000]
```

```
Apr 10 08:24:51 konker2 supervisord[11140]: error: uncaptured python exception, closing channel <supervisor.http.deferring_http_channel connected '' at 0x7f3b0635ef80 channel#: 65 requests:1> (<class 'socket.error'>:[Errno 9] Bad file descriptor [/usr/lib/python2.7/dist-packages/supervisor/medusa/asynchat_25.py|initiate_send|218] [/usr/lib/python2.7/dist-packages/supervisor/medusa/http_server.py|send|512] [/usr/lib/python2.7/dist-packages/supervisor/medusa/asyncore_25.py|send|331] [/usr/lib/python2.7/socket.py|_dummy|174])
```

supervisor.conf (believe it's the standard out of the box conf setup):
```
[unix_http_server]
file=/var/run/supervisor.sock   ; (the path to the socket file)
chmod=0700                       ; sockef file mode (default 0700)

[supervisord]
logfile=/var/log/supervisor/supervisord.log ; (main log file;default $CWD/supervisord.log)
pidfile=/var/run/supervisord.pid ; (supervisord pidfile;default supervisord.pid)
childlogdir=/var/log/supervisor            ; ('AUTO' child log dir, default $TEMP)

; the below section must remain in the config file for RPC
; (supervisorctl/web interface) to work, additional interfaces may be
; added by defining them in separate rpcinterface: sections
[rpcinterface:supervisor]
supervisor.rpcinterface_factory = supervisor.rpcinterface:make_main_rpcinterface

[supervisorctl]
serverurl=unix:///var/run/supervisor.sock ; use a unix:// URL  for a unix socket

; The [include] section can just contain the "files" setting.  This
; setting can list multiple files (separated by whitespace or
; newlines).  It can also contain wildcards.  The filenames are
; interpreted as relative to this file.  Included files *cannot*
; include files themselves.

[include]
files = /etc/supervisor/conf.d/*.conf
```

The program managed by supervisor:
```
[program:queue]
command=php artisan queue:work --tries=1 --timeout=86400 --queue=high,default,low
directory=/var/www/repos/konker
stdout_logfile=/var/www/repos/konker/storage/logs/supervisor.log
redirect_stderr=true
numprocs=8
process_name=%(program_name)s_%(process_num)02d
```
